aks/storage_account/resources.tf

```diff
@@ -0,0 +1,149 @@
+locals {
+  name = var.name != null ? var.name : "default"
+  storage_account_name = var.storage_account_name_override != null ? var.storage_account_name_override : lower(
+    "${var.azure_resource_prefix}${var.service_short}${local.name}${var.config_short}sa"
+  )
+
+  # Determine replication type based on environment
+  effective_replication_type = var.use_production_replication ? var.production_replication_type : var.account_replication_type
```

Comment:
non prod we should force LRS. prod should default to either ZRS or GRS but have the option to override